processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/DefaultJobNotificationExtractor.java

```diff
@@ -0,0 +1,92 @@
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.enumeration.FrequencyType;
+import com.synopsys.integration.alert.common.persistence.accessor.JobAccessor;
+import com.synopsys.integration.alert.common.persistence.model.job.FilteredDistributionJobRequestModel;
+import com.synopsys.integration.alert.common.persistence.model.job.FilteredDistributionJobResponseModel;
+import com.synopsys.integration.alert.processor.api.filter.model.FilterableNotificationWrapper;
+import com.synopsys.integration.blackduck.api.manual.enumeration.NotificationType;
+
+@Component
+public class DefaultJobNotificationExtractor implements JobNotificationExtractor {
+    private JobAccessor jobAccessor;
+
+    @Autowired
+    public DefaultJobNotificationExtractor(JobAccessor jobAccessor) {
+        this.jobAccessor = jobAccessor;
+    }
+
+    /*
+     * Filter Items:
+     * Frequency (Passed into processor)
+     * Notification Type (From notification)
+     * Filter By Project (Projects from notification if applicable)
+     *   Project Name
+     *   Project Name Pattern
+     * Filter by Vulnerability severity (From notification if applicable)
+     * Filter by Policy name (From notification if applicable)
+     */
+
+    @Override
+    public Map<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper<?>>> mapJobsToNotifications(List<? extends FilterableNotificationWrapper<?>> filterableNotifications, @Nullable FrequencyType frequency) {
+        Map<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper<?>>> groupedFilterableNotifications = new HashMap<>();
+
+        for (FilterableNotificationWrapper filterableNotificationWrapper : filterableNotifications) {
+            List<FilteredDistributionJobResponseModel> filteredDistributionJobResponseModels = retrieveMatchingJobs(filterableNotificationWrapper, frequency);
+            for (FilteredDistributionJobResponseModel filteredDistributionJobResponseModel : filteredDistributionJobResponseModels) {
+                List<FilterableNotificationWrapper<?>> set = groupedFilterableNotifications.computeIfAbsent(filteredDistributionJobResponseModel, ignoredKey -> new LinkedList<>());
+                set.add(filterableNotificationWrapper);
+            }
+        }
+        return groupedFilterableNotifications;
+    }
+
+    private List<FilteredDistributionJobResponseModel> retrieveMatchingJobs(FilterableNotificationWrapper filterableNotificationWrapper, @Nullable FrequencyType frequencyType) {
+        // If a frequency is null, we assume they want all frequency types as a frequency is required.
```

Comment:
We should just change the method signature to accept a list of frequencies now.

Reply:
If it's empty then there's nothing to do.

Reply:
I'm thinking of bubbling this all the way up to the NotificationProcessorV2 instead of the processNotifications methods